cfme/common/host_views.py

```diff
@@ -0,0 +1,365 @@
+        row = Text(ParametrizedLocator(".//div[@id='compare-grid']/"
+                                       "/th[normalize-space(.)={drift_section|quote}]"))
+
+        @property
+        def is_changed(self):
+            cells = self.browser.elements(self.CELLS, parent=self.row)
+            attrs = [self.browser.get_attribute("class", cell) for cell in cells]
+            return "drift-delta" in attrs
+
+    @View.nested
+    class toolbar(View):  # noqa
+        all_attributes = Button(title="All attributes")
+        different_values_attributes = Button(title="Attributes with different")
+        same_values_attributes = Button(title="Attributes with same values")
+        details_mode = Button(title="Details Mode")
+        exists_mode = Button(title="Exists Mode")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_compute_infrastructure_hosts and
+            self.title.text == "'{}' Drift Analysis".format(self.context["object"].name)
+        )
+
+
+class HostTimelinesView(TimelinesView, ComputeInfrastructureHostsView):
+    """Represents a Host Timelines page."""
+
+    @property
+    def is_displayed(self):
+        return self.in_compute_infrastructure_hosts and super(TimelinesView, self).is_displayed
+
+
+class HostDiscoverView(ComputeInfrastructureHostsView):
+    """Discover View from Compute/Infrastructure/Hosts page."""
+    esx = Checkbox(name="discover_type_esx")
+    ipmi = Checkbox(name="discover_type_ipmi")
+
+    from_ip1 = Input(name="from_first")
+    from_ip2 = Input(name="from_second")
+    from_ip3 = Input(name="from_third")
+    from_ip4 = Input(name="from_fourth")
+    to_ip4 = Input(name="to_fourth")
+
+    start = Button("Start")
+    cancel = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return self.in_compute_infrastructure_hosts and self.title.text == "Hosts / Nodes Discovery"
+
+
+class HostManagePoliciesView(BaseLoggedInPage):
+    """Host's Manage Policies view."""
+    policies = BootstrapTreeview("protectbox")
+    save = Button("Save")
+    reset = Button("Reset")
+    cancel = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return False
+
+
+class HostEditTagsView(BaseLoggedInPage):
+    """Host's Edit Tags view."""
+    tag_category = BootstrapSelect("tag_cat")
+    tag = BootstrapSelect("tag_add")
+    chosen_tags = Table(locator='.//div[@id="assignments_div"]/table')
+
+    save = Button("Save")
+    reset = Button("Reset")
+    cancel = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return False
+
+
+class HostsToolbar(View):
+    """Represents hosts toolbar and its controls."""
+    configuration = Dropdown(text="Configuration")
+    policy = Dropdown(text="Policy")
+    lifecycle = Dropdown(text="Lifecycle")
+    monitoring = Dropdown(text="Monitoring")
+    power = Dropdown(text="Power")
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class HostItems(View):
```

Comment:
we probably need to use Entities in this case. This is my bug. I lost one of renaming commits during one of rebases :(